NDTensors/test/ITensors/TestITensorDMRG/dmrg.jl

```diff
@@ -0,0 +1,29 @@
+using Test
+using ITensors
+using NDTensors
+using Random
+
+function test_dmrg(elt, N::Integer, dev::Function)
+  sites = siteinds("S=1/2", N)
+
+  os = OpSum()
+  for j in 1:(N - 1)
+    os += "Sz", j, "Sz", j + 1
+    os += 0.5, "S+", j, "S-", j + 1
+    os += 0.5, "S-", j, "S+", j + 1
+  end
+
+  Random.seed!(1234)
+  psi0 = dev(randomMPS(Float64, sites;))
+  H = dev(MPO(elt, os, sites; linkdims=4))
+
+  nsweeps = 3
+  cutoff = [1e-3, 1e-13]
+  noise = [1e-12, 0]
+  ## running these with nsweeps = 100 and no maxdim
+  ## all problems do not have a maxlinkdim > 32
+  maxdim = 32
+
+  energy, psi = dmrg(H, psi0; nsweeps, cutoff, maxdim, noise, outputlevel=1)
+  @test energy ≈ reference_energies[N] rtol = 1e-12
```

Comment:
We need to adjust this for the precision since this won't make sense for single precision. `eps(real(elt))` is a good pattern for that.

Reply:
I usually use something like `10^3 * eps(real(elt))` and then adjust the prefactor until tests pass consistently.